homeassistant/components/stream/worker.py

```diff
@@ -61,12 +35,51 @@ def __init__(self, outputs_callback) -> None:
         self._outputs: list[StreamOutput] = []
         self._sequence = 0
         self._segment_start_pts = None
-        self._stream_buffer = None
+        self._memory_file: BytesIO = cast(BytesIO, None)
+        self._av_output: av.container.OutputContainer = None
+        self._segment: Segment = cast(Segment, None)
+        self._input_video_stream: av.video.VideoStream = None
+        self._input_audio_stream = None  # type=Optional[av.audio.AudioStream]
+        self._output_video_stream: av.video.VideoStream = None
+        self._output_audio_stream = None  # type=Optional[av.audio.AudioStream]
+
+    def reset_av(self) -> None:
```

Comment:
There is now a big of back and forthrequired to follow the logic between `reset_av`, `set_streams`, `reset`, `flush` -- maybe there is some opportunity to combine or re-order a bit.

My first thought is to combine `reset_av` and `reset` into a single method
My second thought is: maybe the av.open call or some of its arguments like container_options should still be called from a helper method.

Or maybe some other approach I haven't thought of.

Reply:
Combining `reset_av` and `reset` is easy if you think that's easier to follow.
Isn't `reset_av` just like the helper method?

Reply:
I was imagining the helper method would not mutate state either by returning the output object (or returning container options), where reset would do the assignment to local variables.  With the container options one, it could also not be a class member and just take arguments.

Reply:
Ok. How about the first way, keeping it in the class but making it a static method?

Reply:
Sure, see if that looks OK to you.